File: java/LinkedListStack.java
public class LinkedListStack {
  private Node top;

  private static class Node {
    int data;
    Node next;

    public Node(int data) {
      this.data = data;
      this.next = null;
    }
  }
  public void push(int data) {
    Node newNode = new Node(data);
    newNode.next = top;
    top = newNode;
  }
  public int pop() {
    if(isEmpty()) {
      System.out.println("stack underflow");
      return -1;
    }
    int data = top.data;
    top = top.next;
    return data;
  }
  public int peek() {
    if(isEmpty()) {
      System.out.println("stack is empty");
      return -1;
    }
    return top.data;
  }
  public boolean isEmpty() {
    return top == null;
  }
}

File: python/linkedlist_stack.py
class Node:
  def __init__(self, data):
    self.data = data
    self.next = None

class LinkedListStack:
  def __init__(self):
    self.top = None

  def push(self, data):
    new_node = Node(data)
    new_node.next = self.top
    self.top = new_node

  def pop(self):
    if self.is_empty():
      raise IndexError("stack underflow")
    item = self.top.data
    self.top = self.top.next
    return item

  def peek(self):
    if self.is_empty():
      raise IndexError("stack is empty")
    return self.top.data

  def is_empty(self):
    return self.top is None
